ethereum/contracts/BasicOutChannel.sol

```diff
@@ -0,0 +1,23 @@
+ // SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "./OutChannel.sol";
+
+// BasicOutChannel is a basic channel that just outputs messages with a nonce.
+contract BasicOutChannel is OutChannel  {
+
+    uint256 public currentNonce;
+
+    constructor() public {
+        currentNonce = 0;
+    }
+
+    /**
+     * @dev Submits a message to the channel
+     */
+    function submit(string memory targetApplicationId, bytes memory payload) public override {
+        emit NewMessage(currentNonce, msg.sender, targetApplicationId, payload);
+        currentNonce++;
```

Comment:
Just checking my understanding... I'm assuming `payload` cannot be replayed here because of the way events are implemented in the EVM?

Reply:
well, the intention here is that this NewMessage will only be accepted by a corresponding InChannel on the parachain side, not by any other other pallet/channel, and that this corresponding InChannel will use the nonce for replay protection. The payload itself/EVM event implementation details are not really part of the replay protection consideration?

Reply:
I'll continue this convo elsewhere. I think my lack of Solidity knowledge is the problem